lib/storage/src/rbac/ops_checks.rs

```diff
@@ -457,3 +452,773 @@ impl PayloadConstraint {
+            op,
+            &AccessCollectionBuilder::new()
+                .add("col", false, true)
+                .into(),
+        );
+
+        assert_allowed_rewrite(
+            op,
+            &AccessCollectionBuilder::new()
+                .add("col", true, false)
+                .into(),
+            |op| {
+                *op = CollectionUpdateOperations::VectorOperation(
+                    VectorOperations::DeleteVectorsByFilter(
+                        make_filter_from_ids(vec![ExtendedPointId::NumId(12345)])
+                            .merge_owned(PayloadConstraint::new_test("col").to_filter()),
+                        vec!["vector".to_string()],
+                    ),
+                );
+            },
+        );
+    }
+
+    /// Tests for [`CollectionUpdateOperations::PayloadOperation`].
+    fn test_collection_update_operations_payload() {
+        for discr in PayloadOpsDiscriminants::iter() {
+            let inner = match discr {
+                PayloadOpsDiscriminants::SetPayload => PayloadOps::SetPayload(SetPayloadOp {
+                    payload: Payload::default(),
+                    points: Some(vec![ExtendedPointId::NumId(12345)]),
+                    filter: None,
+                    key: None,
+                }),
+                PayloadOpsDiscriminants::DeletePayload => {
+                    PayloadOps::DeletePayload(DeletePayloadOp {
+                        keys: vec!["path".parse().unwrap()],
+                        points: Some(vec![ExtendedPointId::NumId(12345)]),
+                        filter: None,
+                    })
+                }
+                PayloadOpsDiscriminants::ClearPayload => PayloadOps::ClearPayload {
+                    points: vec![ExtendedPointId::NumId(12345)],
+                },
+                PayloadOpsDiscriminants::ClearPayloadByFilter => {
+                    PayloadOps::ClearPayloadByFilter(make_filter_from_ids(vec![
+                        ExtendedPointId::NumId(12345),
+                    ]))
+                }
+                PayloadOpsDiscriminants::OverwritePayload => {
+                    PayloadOps::OverwritePayload(SetPayloadOp {
+                        payload: Payload::default(),
+                        points: Some(vec![ExtendedPointId::NumId(12345)]),
+                        filter: None,
+                        key: None,
+                    })
+                }
+            };
+
+            let op = CollectionUpdateOperations::PayloadOperation(inner);
+            assert_requires_whole_write_access(&op);
+        }
+    }
+
+    /// Tests for [`CollectionUpdateOperations::FieldIndexOperation`].
+    fn test_collection_update_operations_field_index() {
+        let op = CollectionUpdateOperations::FieldIndexOperation(
+            FieldIndexOperations::CreateIndex(CreateIndex {
+                field_name: "path".parse().unwrap(),
+                field_schema: None,
+            }),
+        );
```

Comment:
We can also use the same `strum` approach here: we have create and delete operations in the enum